src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java

```diff
@@ -83,57 +101,62 @@ private void endStream(RowStream<Row> stream, RoutingContext ctx,
     });
   }
 
-  Future<Void> getReportTitles(Vertx vertx, RoutingContext ctx) {
-    RequestParameters params = ctx.get(ValidationHandler.REQUEST_CONTEXT_KEY);
-    String tenant = stringOrNull(params.headerParameter(XOkapiHeaders.TENANT));
-    String counterReportId = stringOrNull(params.queryParameter("counterReportId"));
-    String providerId = stringOrNull(params.queryParameter("providerId"));
+  static Future<Void> streamResult(RoutingContext ctx, TenantPgPool pool, String qry,
+                                   String property, Function<Row, JsonObject> handler) {
+    return pool.getConnection().compose(sqlConnection ->
+        sqlConnection.prepare(qry)
+            .<Void>compose(pq ->
+                sqlConnection.begin().compose(tx -> {
+                  ctx.response().setChunked(true);
+                  ctx.response().putHeader("Content-Type", "application/json");
+                  ctx.response().write("{ \"" + property + "\" : [");
+                  AtomicInteger offset = new AtomicInteger();
```

Comment:
`AtomicBoolean first` or `AtomicBoolean comma` is sufficient.

Reply:
yep